Enable usage of HttpOnly Cookies to authenticate browsers (#1959)
kuzzleio/kuzzle: doc/2/api/protocols/http/index.md

HTTP and WebSocket protocols share the same underlying server instance.  
Modifying the listening port will impact these two protocols.
:::

> <SinceBadge version="auto-version"/>

Reviewer: Can be removed: the `pretty` option exists since before kuzzle 1.0

```suggestion
```

HTTP and WebSocket protocols share the same underlying server instance.  
Modifying the listening port will impact these two protocols.
:::

### HTTP-only options

> Only the HTTP protocol does support the following options in a [KuzzleRequest](../../payloads/request/index.md)

Reviewer: We usually use absolute paths in URLs. Also, `index.md` won't exist once the documentation have been generated, since markdowns are used to build HTML files.

When unsure, don't hesitate to test your documentation by running `npm run doc-dev` and by navigating the local copy of the docs on your browser.

```suggestion
Only the HTTP protocol does support the following options in a [KuzzleRequest](/core/2/api/payloads/request)
```

| Property         | Type        | Description                                                                 |
| ---------------- | ----------- | --------------------------------------------------------------------------- |

> | **`cookieOnly`** | **boolean** | **Instruct Kuzzle to store / use authentication token within HTTP Cookies** |

Reviewer: Our documentation style is as follows:
* property/argument names are to be put between quotes
* types are to be put between `pre` tags 

Not really practical for us, but the result is clear and nice :)

```suggestion
| `cookieOnly` | <pre>boolean</pre> | Instructs Kuzzle to store / use authentication tokens within secured HTTP cookies |
```

Reviewer: Also, since you're at it, could you add a documentation for the `pretty` option?

E.g.

```
| `pretty` | <pre>boolean</pre> | Prettify the JSON response, making it more readable for humans |

#### cookieOnly

When the property `cookieOnly` is set to `true` in an HTTP request,
Kuzzle will `store` or `load` the authentication token directly from an [HTTP Cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies),
and will not use the `jwt` property.

::: warning

> The property `cookieOnly` is only supported when the option `supportCookieAuthentication` is enabled in the [Kuzzle Configuration](../../../guides/advanced/configuration/index.md).

Reviewer: Invalid URL:

```suggestion
The property `cookieOnly` is only supported when the option `supportCookieAuthentication` is enabled in the [Kuzzle Configuration](/core/2/guides/advanced/configuration).
```

> Otherwise Kuzzle will treat the request as if `cookieOnly` wasn't set.

Reviewer: ```suggestion
Otherwise Kuzzle will ignore the `cookieOnly` option.
```

HTTP and WebSocket protocols share the same underlying server instance.  
Modifying the listening port will impact these two protocols.
:::

> 

Reviewer 2: Missinge badge `SinceVersion`

### HTTP-only options

Only the HTTP protocol does support the following options in a [KuzzleRequest](/core/2/api/payloads/request)

| Property     | Type               | Description                                                                           |
| ------------ | ------------------ | ------------------------------------------------------------------------------------- |

> | `cookieOnly` | <pre>boolean</pre> | Instructs Kuzzle to store / use authentication tokens within secured HTTP Cookies     |

Reviewer: Sorry for mentioning this only now, but after thinking about it, I don't like the name `cookieOnly`, I don't find it self-explaining.

What about somethiing like `cookieAuth` or something like that? (open to debate obviously)

Reviewer 3: Yep, I like this idea !
Something like `cookieAuth` makes sense, `cookieOnly` makes me think that _sometimes_ cookie **and** sent tokens could be allowed at the same time?

HTTP and WebSocket protocols share the same underlying server instance.  
Modifying the listening port will impact these two protocols.
:::

<SinceBadge version="auto-version"/>
### HTTP-only options

Only the HTTP protocol does support the following options in a [KuzzleRequest](/core/2/api/payloads/request)

| Property     | Type               | Description                                                                           |
| ------------ | ------------------ | ------------------------------------------------------------------------------------- |

> | `cookieAuth` | <pre>boolean</pre> | Instructs Kuzzle to store / use authentication tokens within secured HTTP Cookies     |

Reviewer: well... now that option is only available for a few actions in the `auth` controller (login, checkToken, refreshToken).

So this piece of documentation should be removed from global HTTP options, and instead be added to each one of those routes.

| `pretty`     | <pre>boolean</pre> | Prettify the JSON response, making it more readable for humans                        |

#### cookieAuth

When the property `cookieAuth` is set to `true` in an HTTP request,
Kuzzle will `store` or `load` the authentication token directly from an [HTTP Cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies),
and will not use the `jwt` property.

::: warning
The property `cookieAuth` is only supported when the option `supportCookieAuthentication` is enabled in the [Kuzzle Configuration](/core/2/guides/advanced/configuration).

> Otherwise Kuzzle will ignore the `cookieAuth` option.

Reviewer: Obsolete: this option is now only available if CORS is configured.
